### ISSUE TITLE
codeSearch on GraphCodeBert

### ISSUE THREAD

**Issue by wangsiqidahaoren** [microsoft/CodeBERT#221]
I want to fine tune my own  model on codeSearch on GraphCodeBert, but I found the train_dataset don't have a label(I mean true or flase for wether this codesnippts is related to the NL) as CodeBert like. I want to know if I have some pair of NL-PL data, how can I generate my own train_dataset. What formula will that be.

**Comment by guoday** [comment]
https://github.com/microsoft/CodeBERT/tree/master/GraphCodeBERT/codesearch

Your own dataset is a pickle file, including a list of json data. For each json, you need to include 3 keys: 
'docstring_tokens'  for NL  list of NL tokens
'original_string' for a string of source code
'url' is a unique idx

**Comment by wangsiqidahaoren** [comment]
> 

did that mean I dont need negative samples

**Comment by guoday** [comment]
yes. The negative samples come from other samples in the same batch.

**Comment by wangsiqidahaoren** [comment]
> 

我还想问一下，我对graphcodebert进行微调，是对所有层的权重进行微调，还是只针对后几层进行微调。我是否可以把它看作是一种迁移学习的方案呢

**Comment by guoday** [comment]
所有层。可以
